code/go/0chain.net/smartcontract/zcnsc/stakepool.go

```diff
@@ -43,6 +43,21 @@ func (sp *StakePool) save(sscKey, providerID string, balances cstate.StateContex
 	return
 }
 
+func (sp *StakePool) Save(providerType spenum.Provider, providerID string, balances cstate.StateContextI) (err error) {
+	beforeFunc := func() (e error) {
+		e = sp.StakePool.Save(providerType, providerID, balances)
+		return
+	}
+
+	afterFunc := func() (e error) {
+		_, e = balances.InsertTrieNode(stakepool.StakePoolKey(providerType, providerID), sp)
+		return
+	}
+	
+	err = cstate.WithActivation(balances, "artemis", beforeFunc, afterFunc)
```

Comment:
what will we do with previous records, will we ignore them?

Reply:
We have 20 authorizers on mainnet, each staked with 10000 ZCN so total of 200K ZCN. For now, the simplest fix is to deploy a temporary SC function in the artemis HF that fixes them in a single txn then remove this SC function in the following HF.